overlord/exportstate/manifest.go

```diff
@@ -0,0 +1,209 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020-2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Manifest describes content content exported to snaps or the host.
+type Manifest struct {
+	SnapInstanceName          string        `json:"snap-instance-name"`
+	SnapRevision              snap.Revision `json:"snap-revision"`
+	ExportedForSnapdAsVersion string        `json:"exported-for-snapd-as-version,omitempty"`
```

Comment:
A comment on this field would be helpful (to me, at least :-) )

Reply:
If one of the goals is to support GPU drivers, then we might need to be able to export the content into other snap names than just `snapd` (since I guess that GPU drivers should not appear as being published by snapd?).

So, I suggest to replace this field with a couple of fields:
```
ExportedAsSnapInstanceName string
ExportedAsSnapVersion string
```
Then indeed for the time being we'll have a check that `ExportedAsSnapInstanceName` can only be equal to `snapd` for now, but it's better if we plan the manifest to be generic.

Reply:
After some more thinking: wouldn't it be better to move these fields into the `ExportSet` type? In the future the host system might export not only snapd tools, but also the GPU drivers under another snap name.